core/lib/zksync_core/src/lib.rs

```diff
@@ -287,6 +284,7 @@ impl FromStr for Components {
 }
 
 pub async fn initialize_components(
+    configs: &TempConfigStore,
```

Comment:
A lot of code in this function is suboptimal, e.g. some configs could've been reused instead of cloning, some things could've been refactored along the path, etc.
I opted out of this option since this method is to be removed in future PRs, and spending additional efforts on improving it for a short period of time doesn't look like a good investment.